swap_meet/vendor.py

```diff
@@ -1,2 +1,84 @@
 class Vendor:
-    pass
\ No newline at end of file
+    # Set up inventory
+    def __init__(self, inventory = None):
+        if inventory is None:
+            inventory = []
+        self.inventory = inventory
+        
+    # Adds an item to the inventory
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    # Removes an item from the vendors inventory
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        if item in self.inventory:
+            self.inventory.remove(item)
+        return item
+    
+    # Search through the vendor's inventory for an item with a matching id
+    def get_by_id(self, id):
+        for item in self.inventory:
+            if item.id == id:
+                return item
+        return None
+    
+    #Check both inventories for items, swap items
+    def swap_items(self, other_vendor, my_item, their_item):
+        if my_item not in self.inventory or their_item not in other_vendor.inventory:
+            return False
+        
+        # start swaping
+        self.remove(my_item)
+        other_vendor.add(my_item)
+
+        self.add(their_item)
+        other_vendor.remove(their_item)
+
+        return True
+
+    # Check both inventories, grab first items & swap
+    def swap_first_item(self, other_vendor):
+        if not self.inventory or not other_vendor.inventory:
+            return False
+
+        my_first_item = self.inventory[0]
+        their_first_item = other_vendor.inventory[0]
+
+        return self.swap_items(
+            other_vendor, my_first_item, their_first_item
+        )
```

Comment:
Great use of the `swap_items()` class method here! 

One interesting point to note is that the README never specifies that the items need to be added to the end of each list! The tests don't check for a specific location for each item either. With this in mind, we know we are grabbing the first item from each list, so we could make this a true swap! One option would be to just reassign `self.inventory[0]` to `their_first_item` and `other_Vendor.inventory[0]` to `my_first_item`.

We could also use a Python trick called multiple assignment. This allows us to assign multiple variables to new values all in one line. This might look like:

```suggestion
        self.inventory[0], other_vendor.inventory[0] = other_vendor.inventory[0], self.inventory[0]
        return True
        
```

Reply:
I hadn't thought about using multiple assignment for a “true” swap of list elements, and it’s definitely a much more direct approach.